src/request-manager.ts

```diff
@@ -0,0 +1,27 @@
+type RequestType = 'universalQuery'
```

Comment:
really nitting here, but maybe latestRequestIds can be a type of 
```ts
type RequestIds = {
  universalQuery: number,
  verticalQuery: number, //etc
}
```
and we can have
```ts
type RequestType = keyof RequestIds
```
It's really nitting but there's some chance of there being a typo in the initial keys of `this.latestRequestIds`. Another option is changing latestRequestIds to `Record<RequestType, number>`, though I don't think that would ensure that we are including ALL possible request types